neqo-transport/src/cc/mod.rs

```diff
@@ -0,0 +1,33 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Congestion control
+#![deny(clippy::pedantic)]
+
+pub mod cc;
+mod new_reno_cc;
+
+pub use cc::CongestionControl;
+use new_reno_cc::NewReno;
+pub use new_reno_cc::{CWND_INITIAL_PKTS, CWND_MIN, MAX_DATAGRAM_SIZE, PACING_BURST_SIZE};
+
+pub enum CongestionControlAlgorithm {
+    NewReno,
+}
+
+impl CongestionControlAlgorithm {
+    pub fn create(&self) -> Box<dyn CongestionControl> {
+        match self {
+            Self::NewReno => Box::new(NewReno::default()),
+        }
+    }
```

Comment:
I don't think that it makes sense to structure the creation like this.  Instead, I think that the thing that holds a congestion controller holds `Box<dyn CongestionControl>`, but you do `Box::new(NewReno::new())` to initialize that variable as necessary.

Under what circumstances do you think that we need to have an enum that identifies the type of congestion controller?  The NewReno Debug implementation can say "NewReno" so that logging shows what is going on.  Command-line tools can build their own mappings so that we can pick the CC variant.

Reply:
I want to be able to change congestion control at run time not at compile time. The piece that is missing is API, i.e. Connection::new(..) will have a parameter CongestionControlAlgorithm as well.

I did not want to expose NewReno outside of cc therefore I made this function, but I do not have a strong opinion about this. I will add cubic(because it is easy to add) and the API today and we can discuss about this function whether it make sense or not.

Reply:
This is probably fine then.  One thing is that Connection::new() shouldn't take a CC instance, but a CC factory, because eventually we'll want a new CC for a new path.